### PR TITLE
feat(bigtable): implement LabelFilter

### LINKED ISSUE #1700: bigtable: add more support for cell labels
I was looking through the Bigtable client and noticed there is no support for labelling cells during filtering as well as reading the labels on cells. 

There would be an item in read items with the labels https://godoc.org/cloud.google.com/go/bigtable#ReadItem
As well as a filter to apply labels
https://godoc.org/cloud.google.com/go/bigtable#Filter

### PR DESCRIPTION
**Pull request by IlyaFaer**
Closes #1700

### CODE REVIEW

File: bigtable/filter.go [hunk 1]
```diff
@@ -144,6 +144,18 @@ func (lnf latestNFilter) proto() *btpb.RowFilter {
...
+func ApplyLabelFilter(label string) Filter { return applyLabelFilter(label) }
+
+type applyLabelFilter string
+
+func (alf applyLabelFilter) String() string { return fmt.Sprintf("apply_label(%s)", string(alf)) }
+
+func (alf applyLabelFilter) proto() *btpb.RowFilter {
+	return &btpb.RowFilter{Filter: &btpb.RowFilter_ApplyLabelTransformer{ApplyLabelTransformer: string(alf)}}
+}
+
```
Comment (IlyaFaer) [inline bigtable/filter.go:158]: Adding a labels apply filter given [the limitation](https://github.com/googleapis/go-genproto/blob/053ba62fc06f7e7fd79767598c2e73fbae71e745/googleapis/bigtable/v2/data.pb.go#L1262-L1263)

File: bigtable/reader.go [hunk 2]
```diff
@@ -138,13 +140,15 @@ func (cr *chunkReader) handleCellValue(cc *btpb.ReadRowsResponse_CellChunk) Row
...
 			cr.curVal = make([]byte, 0, cc.ValueSize)
+			cr.curLabels = cc.Labels
 		}
 		cr.curVal = append(cr.curVal, cc.Value...)
 		cr.state = cellInProgress
 	} else {
 		// This cell is either the complete value or the last chunk of a split
 		if cr.curVal == nil {
 			cr.curVal = cc.Value
+			cr.curLabels = cc.Labels
```
Comment (IlyaFaer) [inline bigtable/reader.go:151]: According to [the docs](https://cloud.google.com/bigtable/docs/reference/data/rpc/google.bigtable.v2#cellchunk) labels are sent in the first cell chunk

File: bigtable/reader.go [hunk 3]
```diff
@@ -165,9 +169,11 @@ func (cr *chunkReader) finishCell() {
 		Column:    string(cr.curFam) + ":" + string(cr.curQual),
 		Timestamp: Timestamp(cr.curTS),
 		Value:     cr.curVal,
+		Labels:    cr.curLabels,
 	}
 	cr.curRow[cr.curFam] = append(cr.curRow[cr.curFam], ri)
 	cr.curVal = nil
+	cr.curLabels = nil
```
Comment (IlyaFaer) [inline bigtable/reader.go:176]: Dropping current labels on the cell streaming finish

File: bigtable/integration_test.go [hunk 4]
```diff
@@ -972,6 +982,9 @@ func TestIntegration_Read(t *testing.T) {
 			err := table.ReadRows(ctx, test.rr, func(r Row) bool {
 				for _, ris := range r {
 					for _, ri := range ris {
+						if got, want := ri.Labels, test.wantLabels; !reflect.DeepEqual(got, want) {
+							t.Fatalf("got %q\nwant %q", got, want)
```
Comment (tritone) [inline bigtable/integration_test.go:986]: Is it possible to collect the labels and raise the failure outside the loop? (similar to `elt` on L996)

File: bigtable/reader.go [hunk 5]
```diff
@@ -43,6 +43,7 @@ type ReadItem struct {
 	Row, Column string
 	Timestamp   Timestamp
 	Value       []byte
+	Labels      []string
```
Comment (tritone) [inline bigtable/reader.go:46]: Looks like there are unit tests with ReadItem-- could you add unit tests there which exercise Labels as well?

Comment (IlyaFaer) [reply bigtable/reader.go:46]: Added a test with a splitted cell and a complete cell cases

**Comment by IlyaFaer** [comment]
Friendly ping

File: bigtable/filter.go [hunk 6]
```diff
@@ -144,6 +144,18 @@ func (lnf latestNFilter) proto() *btpb.RowFilter {
 	return &btpb.RowFilter{Filter: &btpb.RowFilter_CellsPerColumnLimitFilter{CellsPerColumnLimitFilter: int32(lnf)}}
 }
 
+// ApplyLabelFilter returns a filter that applies the
+// given label to all cells in the output row.
+func ApplyLabelFilter(label string) Filter { return applyLabelFilter(label) }
```
Comment (codyoss) [inline bigtable/filter.go:149]: nit: should this just be LabelFilter? Might match naming of other filters better?

Comment (codyoss) [reply bigtable/filter.go:149]: cc @tritone thoughts?

Comment (tritone) [reply bigtable/filter.go:149]: Hmm the proto naming also has `Apply` in it, but I'm fine either way. @kolea2 @billyjacobson thoughts?

Comment (billyjacobson) [reply bigtable/filter.go:149]: Let's make it LabelFilter. StripValueFilter is also an apply type of filter and doesn't start with apply. Good catch

Comment (IlyaFaer) [reply bigtable/filter.go]: Done

**Pull request merged**